packages/phishing-controller/src/PhishingDetector.test.ts

```diff
@@ -0,0 +1,979 @@
+            blocklist: [],
+            fuzzylist: [],
+            name: 'second',
+            tolerance: 2,
+            version: 1,
+          },
+        ],
+        async ({ detector }) => {
+          const { result, type, name } = detector.check('allowed-by-both.com');
+
+          expect(result).toBe(false);
+          expect(type).toBe('allowlist');
+          expect(name).toBe('first');
+        },
+      );
+    });
+
+    it('allows a domain when it is in the first fuzzylist and the second allowlist', async () => {
+      await withPhishingDetector(
+        [
+          {
+            allowlist: [],
+            blocklist: [],
+            fuzzylist: ['fuzzy-first-allowed-second.com'],
+            name: 'first',
+            tolerance: 2,
+            version: 1,
+          },
+          {
+            allowlist: ['fuzzy-first-allowed-second.com'],
+            blocklist: [],
+            fuzzylist: [],
+            name: 'second',
+            tolerance: 2,
+            version: 1,
+          },
+        ],
+        async ({ detector }) => {
+          const { result, type, name } = detector.check(
+            'fuzzy-first-allowed-second.com',
+          );
+
+          expect(result).toBe(false);
+          expect(type).toBe('allowlist');
+          expect(name).toBe('second');
+        },
+      );
+    });
+
+    it('allows a domain when it is in the first allowlist and the second fuzzylist', async () => {
+      await withPhishingDetector(
+        [
+          {
+            allowlist: ['allowed-first-fuzzy-second.com'],
+            blocklist: [],
+            fuzzylist: [],
+            name: 'first',
+            tolerance: 2,
+            version: 1,
+          },
+          {
+            allowlist: [],
+            blocklist: [],
+            fuzzylist: ['allowed-first-fuzzy-second.com'],
+            name: 'second',
+            tolerance: 2,
+            version: 1,
+          },
+        ],
+        async ({ detector }) => {
+          const { result, type, name } = detector.check(
+            'allowed-first-fuzzy-second.com',
+          );
+
+          expect(result).toBe(false);
+          expect(type).toBe('allowlist');
+          expect(name).toBe('first');
+        },
+      );
+    });
+
+    it('blocks a blocklisted domain when it ends with a dot', async () => {
+      await withPhishingDetector(
+        [
+          {
+            allowlist: [],
+            blocklist: ['blocked.com'],
+            fuzzylist: [],
+            name: 'first',
+            tolerance: 2,
+            version: 1,
+          },
+        ],
+        async ({ detector }) => {
+          const { result, type } = detector.check('blocked.com.');
+
+          expect(result).toBe(true);
+          expect(type).toBe('blocklist');
+        },
+      );
+    });
+
+    describe('with legacy config', () => {
+      it('changes the type to whitelist when the result is allowlist', async () => {
```

Comment:
It appears (based on the names of these tests) that we are treating the legacy API as an anomaly instead of placing it on equal footing as the recommended API.

We don't need to test that the detector works with multiple config objects, as you can't do that with the legacy API of course, but what are your thoughts on testing what happens if the domain is matched via fuzzylist? Or not matched via the fuzzylist?

Reply:
These test cases are meant to check if `allowlist` and `blocklist` are projected into `whitelist` and `blacklist` for legacy configs, but this is not the case with fuzzylists, which should always be reported as `fuzzy` also with legacy configs. 

We can add a test case like this anyway though:

```typescript
      it('uses the type `fuzzy` when the result is in fuzzylist', async () => {
        await withPhishingDetector(
          {
            blacklist: [],
            fuzzylist: ['fuzzy.com'],
            tolerance: 2,
            whitelist: [],
          },
          async ({ detector }) => {
            const { type, result } = detector.check('fupzy.com');

            expect(type).toBe('fuzzy');
            expect(result).toBe(true);
          },
        );
      });
```

Reply:
Ah I see, makes sense. That sounds like a good test to add.

Reply:
Added in [015fbd0](https://github.com/MetaMask/core/pull/4137/commits/015fbd0724255041953580473a490ce182b20945)